src/Telegram.php

```diff
@@ -396,32 +389,31 @@ public function handle()
      *
      * @return \Longman\TelegramBot\Telegram
      */
-
-    public function processUpdate(update $update)
+    public function processUpdate(Update $update)
     {
         $update_type = $update->getUpdateType();
         if ($update_type == 'message') {
             //Load admin Commands
             if ($this->admin_enabled) {
                 $message = $update->getMessage();
-    
+
                 //Admin command avaiable in any chats
                 //$from = $message->getFrom();
                 //$user_id = $from->getId();
-    
+
                 //Admin command avaiable only in single chat with the bot
                 $chat = $message->getChat();
                 $user_id = $chat->getId();
-    
+
                 if (in_array($user_id, $this->admins_list)) {
-                    $this->addCommandsPath(BASE_PATH.'/Admin');
+                    $this->addCommandsPath(BASE_PATH . '/Admin');
                 }
             }
-    
+
             // check type
             $message = $update->getMessage();
             $type = $message->getType();
-    
+
             switch ($type) {
```

Comment:
It would be nice shrink!
But let me explain some issue to overcome!

There are two kind of commands:
- Command about chat trigger as new_chat_participant...
- Commands about type of message as Video, Voice...

Each one belong to first type has is own command, instead the latter type is completely redirected to Generic.
I wonder if can be a  good idea to separate those two kind of commands in this way:

##Current configuration
Message entity has [type](https://github.com/akalongman/php-telegram-bot/blob/master/src/Entities/Message.php#L75) member that can be: new_chat_participant, ... , command, Video, .. , Message ecc.

##New configuration
Message entity with two new member:
- *trigger*: new_chat_participant, Message, Command, Left_chat_participant, (all the chat trigger + command and message )
- *message_type*: Video, Photo, text, Sticker, null if *trigger* is  not Message or Command

In this way we can have a command for each *trigger* and we can handle different type of messages inside the GenericmessageCommand.
What do you think? 

Ah! Of course we can directly return the command result! :)